src/pyhf/pdf.py

```diff
@@ -222,12 +222,22 @@ def __init__(self, spec, **config_kwargs):
             spec, self.channel_nbins
         )
 
-        poiname = config_kwargs.get('poiname', 'mu')
+        poiname = config_kwargs.pop('poiname', 'mu')
+
+        # measurement_name is explicitly removed in cases where the model is created using poi_name, multiple
+        # measurements are there but the first one is taken or if measurement is defined as None in the argument
+        if 'measurement_name' in config_kwargs:
+            del config_kwargs['measurement_name']
+
         default_modifier_settings = {'normsys': {'interpcode': 'code1'}}
-        self.modifier_settings = (
-            config_kwargs.get('modifier_settings') or default_modifier_settings
+        self.modifier_settings = config_kwargs.pop(
+            'modifier_settings', default_modifier_settings
         )
 
+        assert (
```

Comment:
In general I think it is a good idea to avoid the use of `assert` outside of tests, as [they can be removed unintentionally](https://stackoverflow.com/a/1838411/8931942). It would be better to do a check here and then raise the proper error.